contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.0.0--3.1.0.sql

```diff
@@ -1100,6 +1100,109 @@ AND sys.babelfish_get_pltsql_function_signature(ao.object_id) = f.funcsignature
 WHERE ao.type in ('P', 'RF', 'V', 'TR', 'FN', 'IF', 'TF', 'R');
 GRANT SELECT ON sys.all_sql_modules_internal TO PUBLIC;
 
+-- deprecate old FOR XML/JSON functions if they exist - if this is a fresh install of 3.1 they won't exist, so check for that
+DO $$
+BEGIN IF (SELECT count(*) FROM pg_proc as p where p.proname = 'tsql_query_to_xml') = 1 THEN
```

Comment:
This condition could be false in the following cases:
- when a customer creates a function with the same name in non-sys schema
- when customer created a function with the same name and a different signature in sys schema

If deprecating the old function is not mandatory, could you just use TRY/CATCH?

Reply:
good point, I've changed these blocks to use BEGIN...EXCEPTION